scipy/optimize/_remove_redundancy.py

```diff
@@ -318,6 +351,8 @@ def _remove_redundancy_sparse(A, rhs):
             e[i-1] = 0
 
         pi = scipy.sparse.linalg.spsolve(B.transpose(), e).reshape(-1, 1)
+        pi2 = B2.solve(e, True)
+        assert(np.allclose(pi.ravel(), pi2.ravel()))
```

Comment:
I'm using the `LUSolve` object in parallel with the original code just to check accuracy. The `assert` checks that the `LUSolve` object is working correctly.